File: basemodule/src/main/java/zjm/cst/dhu/basemodule/view/BaseActivity.java
package zjm.cst.dhu.basemodule.view;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;


/**
 * Created by zjm on 2017/2/24.
 */

public abstract class BaseActivity extends AppCompatActivity {

    protected abstract void injectDependences();

    protected abstract void dataBinding();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        injectDependences();
    }

    @Override
    protected void onStart() {
        super.onStart();
    }

    @Override
    protected void onStop() {
        super.onStop();
    }
}

File: basemodule/src/main/java/zjm/cst/dhu/basemodule/view/BaseActivity.kt
package zjm.cst.dhu.basemodule.view

import android.os.Bundle
import android.support.v7.app.AppCompatActivity


/**
 * Created by zjm on 2017/2/24.
 */

abstract class BaseActivity : AppCompatActivity() {

    protected abstract fun injectDependences()

    protected abstract fun dataBinding()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        injectDependences()
    }

    override fun onStart() {
        super.onStart()
    }

    override fun onStop() {
        super.onStop()
    }
}
